go/vt/vtgate/planbuilder/operators/route_planning.go

```diff
@@ -63,6 +62,28 @@ func transformToPhysical(ctx *plancontext.PlanningContext, in ops.Operator) (ops
 			return optimizeSubQuery(ctx, op)
 		case *Filter:
 			return optimizeFilter(op)
+		case *Horizon:
+			union, ok := op.Source.(*Union)
+			if !ok {
+				return op, rewrite.SameTree, nil
+			}
+
+			var newSrcs []ops.Operator
+			for _, src := range union.Sources {
+				_, ok = src.(*Route)
+				if !ok {
+					return op, rewrite.SameTree, nil
+				}
+
+				// if we got here we are dealing with a UNION created for an info_schema table
```

Comment:
I don't quite understand this. Why is it necessary that the union is for `info_schema` tables. Could it not just a union of other routes?